speechbrain/utils/distributed.py

```diff
@@ -116,19 +116,42 @@ def run_on_main(
             ddp_barrier()
 
 
-def if_main_process():
-    """Checks if the current process is the main process and authorized to run
-    I/O commands. The main process is the one with `RANK == 0`. In standard mode,
-    the process will not have `RANK` Unix var and will be authorized to run the I/O commands.
+def is_distributed_initialized() -> bool:
+    "Returns whether the current system is distributed."
+    # `is_initialized` is only defined conditionally
+    # https://github.com/pytorch/pytorch/blob/v2.1.0/torch/distributed/__init__.py#L25
+    # this might happen to MacOS builds from source (default) or any build from source that sets `USE_DISTRIBUTED=0`
+    return (
+        torch.distributed.is_available() and torch.distributed.is_initialized()
+    )
+
+
+def if_main_process() -> bool:
+    "Returns whether the current process is the main process."
+    if is_distributed_initialized():
+        return torch.distributed.get_rank() == 0
+    else:
+        return True
+
+
+class MainProcessGuard:
```

Comment:
I'm a little unsure about the name `MainProcessGuard` because this context manager only manages tracking state rather than guarding anything. Would `MainProcessContext` or something make more sense?

Reply:
I agree. I was a bit puzzled by the name and I didn't knew which one to use. I prefer yours. Thanks.